uniflow/op/model/model_server.py

```diff
@@ -474,3 +476,201 @@ def __call__(self, data: List[str]) -> List[str]:
                         out = re.sub(r"\n{3,}", "\n\n", out).strip()
             outs.append(out)
         return outs
+
+class BedrockModelServer(AbsModelServer):
+    """Bedrock Model Server Class.
+    
+    The AWS client authenticates by automatically loading credentials as per the methods outlined here:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
+
+    If you wish to use a specific credential profile, please provide the profile name from your ~/.aws/credentials file.
+
+    Make sure that the credentials or roles in use have the necessary policies for Bedrock service access.
+
+    Additionally, it is important to verify that your boto3 version supports the Bedrock runtime.
+    """
+
+    def __init__(self, model_config: Dict[str, Any]) -> None:
+
+        try:
+            # import in class level to avoid installing boto3
+            import boto3
+
+            super().__init__(model_config)
+            self._model_config = BedrockModelConfig(**self._model_config)
+
+            aws_profile = self._model_config.aws_profile
+            aws_region = self._model_config.aws_region if self._model_config.aws_region else None
+            session = boto3.Session(profile_name=aws_profile)
+
+            self._client = session.client("bedrock-runtime", region_name=aws_region)
+
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Failed to import the 'boto3' Python package. "
+                "Please install it by running `pip install boto3`."
+            )
+        except Exception as e:
+            raise ValueError(
+                "Failed to load credentials for authenticating with the AWS client. "
+                "Please ensure that the specified profile name contains valid credentials."
+            ) from e
+
+    def _preprocess(self, data: List[str]) -> List[str]:
+        """Preprocess data.
+
+        Args:
+            data (List[str]): Data to preprocess.
+
+        Returns:
+            List[str]: Preprocessed data.
+        """
+        return data
+
+    def _postprocess(self, data: List[str]) -> List[str]:
+        """Postprocess data.
+
+        Args:
+            data (str): Data to postprocess.
+
+        Returns:
+            List[str]: Postprocessed data.
+        """
+        return data
+    
+    def _get_provider(self) -> str:
+        return self._model_config.model_name.split(".")[0]
+    
+    def enforce_stop_tokens(text: str, stop: List[str]) -> str:
+        """Cut off the text as soon as any stop words occur."""
+        return re.split("|".join(stop), text, maxsplit=1)[0]
+    
+    def prepare_input(
+        self, provider: str, prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
```

Comment:
nit: a docstring will be really helpful to increase readability regarding `prepare_input` is use provider to format model input prompt.

Reply:
Will fix it right away

Reply:
Fixed